cli/azd/internal/telemetry/appinsights-exporter/transmitter.go

```diff
@@ -1,4 +1,6 @@
-package appinsights
+package appinsightsexporter
+
+// Forked implementation from github.com/microsoft/ApplicationInsights-Go
```

Comment:
I don't want to block progress here, and I'm fine accumulating some debt for now to get telemetry working - but I think we need to come up with a long-term plan for how we manage this this dependency.

I think our general strategy for telemetry (batching items and sending later, vs a real time stream) makes sense for console apps like ours and it's unfortunate we have to hack the SDK like this.  I wonder if the maintainers of the library would be open to exposing the transmitter stuff publicly at some point?

One small tactical concern is how we maintain this code going forward.  Since we squash merge everything, we're going to lose understanding of the changes you made and so tracking updates will be difficult.

I'm honestly not sure what the right thing for us to be doing here is - maybe it means forking the library (like an actual fork) with a branch that has your changes, so we can use git to manage merging in changes from upstream, and then using a `go.mod` override to point at the fork?

You might also consider taking some of the net new methods (like TelemetryItems::Deserialize) and place them in separate files, just to make it easier to compare our copy to upstream.  Additionally, it might be nice to include in the comments of these files the commit hash you used when you synced and small notes about the changes you made (exported TelemetryItems, replaced `diagnosticsWriter` with an implementation from the `log` package, etc).

That way when we go to update these files we won't have to rely on this PR's history to understand the changes.

Along similar lines, I would personally minimize things like moving from `ioutil` or fixing lint errors - let's make the minimum number of edits to the code we can (we can just suppress the lint warnings in these files) to minimize drift from upstream.

Reply:
Oops, I forgot to mention this -- [appinsights-go](https://github.com/microsoft/ApplicationInsights-Go) is no longer maintained: so there wouldn't be any upstream updates we'd worry about. Also, the small piece we're reusing is the HTTP client that does a single POST request to a single API.

Spiritually, this is more of an implementation re-use of a single piece of code than forking of a library. I could have spent time implementing the same HTTP client but would end up with almost 90% of the same code as the current implementation.

Also, the larger plan here is for us to potentially release the `appinsights-exporter` library as a counterpart to `appinsights-go`. An OpenTelemetry direct exporter for Azure Monitor. The groundwork is laid down in these telemetry changes, but due to special requirements for our CLI, they don't directly align and so there are missing pieces unimplemented.

Reply:
> Oops, I forgot to mention this -- [appinsights-go](https://github.com/microsoft/ApplicationInsights-Go) is no longer maintained: so there wouldn't be any upstream updates we'd worry about. 

Ahh - all right - I feel much less nervous about this whole thing then.

What you're doing makes a bunch of sense.